60_network_learning.Rmd

```diff
@@ -0,0 +1,654 @@
+
+```{r, message=FALSE, warning=FALSE}
+library(corpcor)
+```
+
+```{r}
+# norm_counts: clr-transformed count matrix with samples in rows
+rho_shrink_est <- function(norm_counts, ...) {
+  shrunk_cov <- cov.shrink(norm_counts, ...)
+  p <- ncol(norm_counts)
+  J <- matrix(rep(diag(shrunk_cov), p), p)
+  rho <- 2 * shrunk_cov / (J + t(J))
+  (rho + t(rho)) / 2
+}
+```
+
+```{r compute_shrink_prop}
+# Apply the shrinkage proportionality estimator to the clr-transformed counts
+prop_est <- as(rho_shrink_est(t(assay(tse, "clr"))), "matrix")
+```
+
+Again, we use our transformation function to convert the association matrix into a graph object.
+
+```{r transform_shrink_prop}
+prop_trans <- transform_asso(prop_est, thresh = 0.4)
+prop_graph <- prop_trans$graph
+```
+
+### SpiecEasi - MB {#spieceasi-mb}
+
+The third association measure introduced is partial correlation (or conditional dependence). We use the SpiecEasi ("Sparse InversE Covariance estimation for Ecological Association and Statistical Inference") approach proposed by @Kurtz2015 to estimate a sparse correlation matrix. The neighborhood selection method ("MB") introduced by @meinshausen2006high is used for network learning. The approach is implemented in the R package [SpiecEasi](https://github.com/zdk123/SpiecEasi). 
+
+```{r, message=FALSE, warning=FALSE}
+library(SpiecEasi)
+```
+
+```{r, eval=FALSE}
+set.seed(13075)
+se_mb_est <- spiec.easi(t(assay(tse, "counts")), 
+                        method = 'mb', nlambda = 20, 
+                        pulsar.params = list(rep.num = 20))
+```
+
+Since `SpiecEasi` uses the StARS ("Stability Approach to Regularization Selection") method [@liu2010stability] to obtain a sparse association matrix, we don't need to set a threshold here. We store the partial correlations corresponding to the StARS-optimal lambda and convert them into an `igraph` object.
+
+```{r transform_se_mb}
+# Get optimal matrix with partial correlations
+se_mb_cor <- as.matrix(getOptBeta(se_mb_est))
+se_mb_cor <- as.matrix(symBeta(se_mb_cor))
+rownames(se_mb_cor) <- colnames(se_mb_cor) <- rownames(tse)
+diag(se_mb_cor) <- 1
+
+# Create graph object
+se_mb_graph <- transform_asso(se_mb_cor)$graph
+```
+
+### Network plots
+
+The graph objects can now be plotted using the [igraph](https://r.igraph.org/) package, which is a state-of-the-art package for network analysis and visualization. Other packages that could be used for network plotting are the [qgraph](https://rdrr.io/cran/qgraph/) package or the [ggnet2](https://briatte.github.io/ggnet/) package. Since we are going to use igraph for network analysis later on, we are using its plotting function here as well.
+
+The four graph objects are plotted side-by-side using the same layout so that the networks are comparable.
+
+```{r, message=FALSE, warning=FALSE}
+library(igraph)
+```
+
+```{r nets_sparcc_prop_se, fig.height=10, fig.width=10}
+# Node sizes
+vsize <- colMeans(t(assay(tse, "clr"))) + 6
```

Comment:
I am not sure if mean over CLR is meaningful measure for node size? CLR can include both neg and pos values?